src/v/raft/state_machine_base.h

```diff
@@ -42,7 +42,7 @@ class state_machine_base {
       model::offset,
       model::timeout_clock::time_point,
       std::optional<std::reference_wrapper<ss::abort_source>> as
-      = std::nullopt);
+      = std::nullopt) const;
```

Comment:
looks a little odd to be const since it literally means wait until the state machine state changes to this offset :D

Reply:
actually may be not, the caller probably only waits for the action to finish while the action that mutates the state is done by something else (as in the commit message).